plugins/woocommerce/tests/php/src/Caches/Invalidators/ProductCacheInvalidatorTest.php

```
@@ -0,0 +1,184 @@
+
+		$this->assertNotEmpty( $this->captured_actions );
+		$this->assertEquals( $product->get_id(), $this->captured_actions[0]['product_id'] );
+		$this->assertEquals( CacheInvalidatorInterface::OPERATION_UPDATE, $this->captured_actions[0]['operation'] );
+	}
+
+	/**
+	 * @testdox Creating a variable product with variations fires appropriate invalidations.
+	 */
+	public function test_variable_product_creation() {
+		$parent_product = \WC_Helper_Product::create_variation_product();
+		$variations     = $parent_product->get_children();
```

PHPMD (2.15.0)

Avoid unused local variables such as '$variations'. (undefined)

(UnusedLocalVariable)